lib/kokkos/core/src/Kokkos_Macros.hpp

```diff
@@ -653,7 +653,9 @@ static constexpr bool kokkos_omp_on_host() { return false; }
 #if (defined(KOKKOS_COMPILER_GNU) || defined(KOKKOS_COMPILER_CLANG) ||  \
      defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_PGI)) && \
     !defined(_WIN32)
+#if (defined(__linux__) && defined(__GLIBC_MINOR__))
```

Comment:
@akohlmey should this be ported to upstream Kokkos?

Reply:
Good question. I don't have an answer for that, only some explanation of why I made this change.

The proper solution is already in their version 4.0 branch, but that requires C++17.
I read a comment from @crtrott that they want to support the current version with bugfixes for some time, so they may be interested to include this. But I am using some kind of "inofficial heuristic" here to tell GNU libc apart from musl-libc since the latter does not define a macro that defines it, so there is a small chance that it would be the right heuristic for all Kokkos use cases. The musl-libc is Linux-only and rarely used outside of embedded system environments. But you *can* configure Arch and Gentoo Linux to use it instead of GNU libc, hence the bug-reports (I found one on the Arch website) and the (proper) fix in Kokkos 4.

My use case is rather exotic: I am using a Linux-to-Linux cross-compiler that targets building fully binary portable and statically linked non-MPI executables for LAMMPS and the LAMMPS shell program (which is not really possible in the same way with GNU libc). The main intent is to offer people a download to quickly run a fully patched binary of LAMMPS (stable or develop branch) without having to compile it themselves for testing if a bug can be reproduced with the respective current version. But this can also be a more lightweight alternative to distributing virtual machine images for LAMMPS tutorials. Or useful for people that just want to try out a few things with LAMMPS.

Reply:
@crtrott @dalg24 FYI, see change and discussion above for Kokkos that may need to be ported v3.7.2